osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/messaging/RequestWithMetadata.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright 2021 Alliander N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging;
```

Comment:
This class has been added to the code of the protocol adapter. It does not appear to have anything specific to the protocol adapter though. It is useful for code that needs to do something with the message metadata in handling messages.
Since the secret management service is intended for use in other situations than the DLMS protocol adapter, and handling metadata with the request could become desirable in other scenarios as well, it looks like it would be a good idea to put this code in one of the shared projects, so it is available outside the protocol adapter.

Reply:
Moved class to `package org.opensmartgridplatform.shared.infra.jms` in shared project